### DISCUSSION TITLE
Accessing wrong URL results in not loading the correct translation file

### DISCUSSION THREAD

**Discussion by moyounishimself** [vinissimus/next-translate#1221] [Q&A]
Hi there,

I'm using Next 14 with page routing and recently introduced a locale so now I have `de` and `en`. One issue I noticed is that when I access a wrong url the locale isn't loaded.

Steps to reproduce:

1. open `localhost/this-page-does-not-exist`
2. 404 page UI should render with user locale
3. Try switching locale using `setLanguage` and now it can't find the translation

https://github.com/user-attachments/assets/34eb970b-0f51-4861-b92c-21c75e5b90b0

It tries to load `{locale}/this-page-does-not-exist.json` for some reason. Do you have any idea how to fix this? Aside from redirecting to 404 url

**Comment by moyounishimself** [comment]
Any suggestions? @aralroca

**Reply by aralroca** [reply]
On pages router you can use `/404` to load the namespaces: https://github.com/aralroca/next-translate/blob/e6361250a817672e9012e7b1973483a18e421fd3/examples/complex/i18n.js#L7

On app router is similar but with `/_not-found`
